### PR TITLE
GitHub repo structure

### PR DESCRIPTION
**Pull request by rodolfocarobene**
This PR gives a bit more structure to the repository and implements poetry installation (solving #3)

### CODE REVIEW

File: fpga-server/src/rfsoc_server.py
[the remark this answers is not in the document]
Comment (scarrazza) [reply fpga-server/src/rfsoc_server.py:1]: Done, please update the README.md as soon as possible.

**Comment by rodolfocarobene** [comment]
I think that this PR could and should already be merged. If something is wrong we can correct it later, but having a good structure will help in solving the other issues (#5 #4).
Also, the updated readme would be on main

**Comment by AleCandido** [comment]
> I think that this PR could and should already be merged. If something is wrong we can correct it later, but having a good structure will help in solving the other issues (#5 #4).
> Also, the updated readme would be on main

If you believe so, remove draft mode :P

**Comment by AleCandido** [comment]
`git rebase new_rfsoc_server`

**Review by AleCandido — APPROVED** [review]
After rebasing, it can be merged.

**Comment by AleCandido** [comment]
However, notice that merging before #2 is slightly inconvenient: this PR targets another branch, not `main`, so we will get all these files back to review again.

If possible, it would be better to rebase on `main` and merge there, inverting the dependency between this and #2 (i.e. rebasing #2 on this one).

However, for this specific case changes are well-separated by file, so we could also skip them in #2 while reviewing (and #2 only impacts one file).

**Pull request merged**